### DISCUSSION TITLE
What is the relationship between the handler config in the main config files and rules?

### DISCUSSION THREAD

**Discussion by cameron-martin** [ory/oathkeeper#660] [Q&A] [answered]
There are two ways of specifying handlers, both in the main config file and the access rules files. However, it is unclear to me how these interact. For example:

* What happens if the handler is not enabled in the main config file but is specified in the rules?
* If a handler is specified in the main config file, is it automatically enabled or does it also have to also be specified in the rules?
* If a handler is specified in the main config file and also in the rules, how does the config for these interact?

**Comment by Benehiko** [comment, marked as answer]
Hi @cameron-martin,

Sorry for the late reply! We seemed to have missed your question somehow :cold_sweat:

Do you mean in relation to something like this? https://www.ory.sh/oathkeeper/docs/pipeline/mutator/#configuration-2

1. The global config file needs to enable the specified handler at least before it can be used inside the access rules file.
2. It is globally enabled for any access rule
3. The access rule file overrides any global defined configuration.

**Reply by Benehiko** [reply]
Hi @cameron-martin,

Has your issue been resolved?

**Reply by cameron-martin** [reply]
Yes, sorry I forgot to reply. I'll mark your answer as accepted.
